datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -552,6 +552,22 @@ make_data_page_stats_iterator!(MinInt32DataPageStatsIterator, min, Index::INT32,
 make_data_page_stats_iterator!(MaxInt32DataPageStatsIterator, max, Index::INT32, i32);
 make_data_page_stats_iterator!(MinInt64DataPageStatsIterator, min, Index::INT64, i64);
 make_data_page_stats_iterator!(MaxInt64DataPageStatsIterator, max, Index::INT64, i64);
+make_data_page_stats_iterator!(
+    MinFloat16DataPageStatsIterator,
+    min,
+    Index::FIXED_LEN_BYTE_ARRAY,
+    f16
+);
+make_data_page_stats_iterator!(
+    MaxFloat16DataPageStatsIterator,
+    max,
+    Index::FIXED_LEN_BYTE_ARRAY,
+    f16
+);
+make_data_page_stats_iterator!(MinFloat32DataPageStatsIterator, min, Index::FLOAT, f32);
```

Comment:
I think `Float` (f32) and `Double` (f64) would be enough, in terms of naming. This would also be consistent with the row group stats iterators.